### PR TITLE
run-android-instrumentation-tests 1.0.0

### PR DESCRIPTION
**Pull request by tauntz**
TagCheck

### What to do if the build fails?

At the moment contributors do not have access to the CI workflow triggered by StepLib PRs. In case of a failed build, we ask for your patience. Maintainers of Bitrise Steplib will sort it out for you or inform you if any further action is needed.

### New Pull Request Checklist

*Please mark the points which you did / accept.*

- [x] __I will not move an already shared step version's tag to another commit__
- [x] I read the [Step Development Guideline](https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md)
- [x] I have a test for my Step, which can be run with `bitrise run test` (in the step's repository)
- [x] I did run `bitrise run audit-this-step` (in the step's repository - note, if you don't have this workflow in your `bitrise.yml`, [you can copy it from the step template](https://github.com/bitrise-steplib/step-template/blob/master/bitrise.yml).)
- [x] I read and accept the [Abandoned Step policy](https://github.com/bitrise-io/bitrise-steplib#abandoned-step-policy)

**New Step**
Thank you for the new Step share! The CI check might will fail due to our extended validation engine. Nothing to worry about yet, we will get back to you shortly.

### CODE REVIEW

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
+title: Run instrumentation tests
```
Comment (koral--) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:1]: This name (and step id) suggests rather running instrumentation tests on devices/emulators connected to the bitrise VM via adb than using some external service.
The name should be "Run instrumentation tests on emulator.wtf" or something similar.

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    - "no"
+  with_coverage: "no"
+- opts:
+    summary: Using $BITRISE_TEST_RESULT_DIR here will make sure that they are also
+      available under the Test Reports view
+    title: Output path where test run outputs (junit XML, logcat, files) will be stored.
+  outputs_dir: $BITRISE_TEST_RESULT_DIR
+- additional_apks: null
+  opts:
+    summary: A comma separated list of additional APKs to install, in addition to
```
Comment (koral--) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:112]: Is it list of local file paths, urls, both are supported?

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+host_os_tags:
+- osx-10.10
+- ubuntu-16.04
+project_type_tags:
+- android
+type_tags:
+- test
+toolkit:
+  bash:
+    entry_file: step.sh
```
Comment (koral--) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:26]: That script downloads a binary from external server at: https://github.com/emulator-wtf/run-tests-bitrise/blob/main/step.sh#L4
That is against step development guideline:
https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md#do-not-use-submodules-or-require-any-other-resource-downloaded-on-demand

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+  outputs_dir: $BITRISE_TEST_RESULT_DIR
+- additional_apks: null
+  opts:
+    summary: A comma separated list of additional APKs to install, in addition to
+      those being directly tested.
+    title: Additional APKs to install
+- environment_variables: null
+  opts:
+    description: |
+      Comma separated list of key=value pairs that are passed as extra options to the `am instrument` command.
```
Comment (koral--) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:118]: Pipe should be used as a separator, not the comma: https://github.com/bitrise-io/bitrise/blob/master/_docs/step-development-guideline.md#inputs-which-can-accept-a-list-of-values

File: steps/run-android-instrumentation-tests/1.0.0/step.yml (hunk 1 above)
Comment (mateherber) [reply steps/run-android-instrumentation-tests/1.0.0/step.yml:1]: Also please update the id of the step in step library from `run-android-instrumentation-tests` to something like `emulator-wtf-android-test`

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+project_type_tags:
+- android
+type_tags:
+- test
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: curl
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:29]: According to @koral-- 's review comment this dependency will no longer be needed.

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+
+  It's like Firebase Test Lab on steroids (2-4x speedup, depending on your specific setup)
+website: https://emulator.wtf
+source_code_url: https://github.com/emulator-wtf/run-tests-bitrise
+support_url: https://github.com/emulator-wtf/run-tests-bitrise/issues
+published_at: 2021-11-28T17:10:15.656951+02:00
+source:
+  git: https://github.com/emulator-wtf/run-tests-bitrise.git
+  commit: b4e5502f25bbec6c9b1db6f7f519b0361b293ad3
+host_os_tags:
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:17]: Please remove `host_os_tags` as it is no longer used

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+- test
+toolkit:
+  bash:
+    entry_file: step.sh
+deps:
+  brew:
+  - name: curl
+  apt_get:
+  - name: curl
+is_requires_admin_user: true
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:32]: Please remove `is_requires_admin_user` as it is not used anymore

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  use_orchestrator: "yes"
+- clear_package_data: "yes"
+  opts:
+    summary: Only applicable when using Android Test Orchestrator
+    title: Clear package data between runs
+    value_options:
+    - "yes"
+    - "no"
+- opts:
+    title: Collect code coverage data while executing the test
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:100]: Please add `summary` to this input. (All inputs should have title and summary at least, while description is optional).

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 9]
```diff
@@ -0,0 +1,136 @@
...
+    summary: Using $BITRISE_TEST_RESULT_DIR here will make sure that they are also
+      available under the Test Reports view
+    title: Output path where test run outputs (junit XML, logcat, files) will be stored.
+  outputs_dir: $BITRISE_TEST_RESULT_DIR
+- additional_apks: null
+  opts:
+    summary: A comma separated list of additional APKs to install, in addition to
+      those being directly tested.
+    title: Additional APKs to install
+- environment_variables: null
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:115]: please use the `environment_variable_list` naming convention on inputs where you have multiple values separated by pipe

File: steps/run-android-instrumentation-tests/1.0.0/step.yml [hunk 10]
```diff
@@ -0,0 +1,136 @@
...
+
+      Example use-cases:
+        * Run second shard out of 3 shards: `numShards=3,shardIndex=1`
+        * Run tests in a specific class: `class=com.example.app.MyTestClass`
+    summary: Extra options are passed as extra options to the `am instrument` command.
+    title: Test extra options
+- directories_to_pull: null
+  opts:
+    description: |
+      Comma separated directories that should be pulled from the emulator and that will be stored in the output path together with tests results.
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/1.0.0/step.yml:130]: Same not to separation and input id name's convention as above

File: steps/run-android-instrumentation-tests/assets/icon.svg [hunk 11]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8" standalone="no"?>
+<svg id="emulator.wtf" viewBox="0 0 256 256" version="1.1" width="256" height="256" xmlns="http://www.w3.org/2000/svg" xmlns:svg="http://www.w3.org/2000/svg">
+  <defs id="defs137">
+    <style id="style135">
+      .cls-1{fill:#212020;}.cls-2{fill:#8fe2aa;}.cls-3{fill:#67cdcc;}
+    </style>
+  </defs>
+  <rect class="cls-1" x="0" y="0" width="256" height="256" rx="22.099592" id="rect139" style="stroke-width:1.0449"/>
```
Comment (mateherber) [inline steps/run-android-instrumentation-tests/assets/icon.svg:8]: Please do not use rounded corners for the icon, it will not look good.

**Review by mateherber — CHANGES_REQUESTED** [review]
Thank you for your contribution, I've added my requests.

**Pull request closed without merging**